src/writer.js

```diff
@@ -185,6 +179,29 @@ define([
         });
     };
 
+    var createSetValues = function (dataTree, form, xmlWriter) {
+        function writeSetValue(setValue) {
+            xmlWriter.writeStartElement('setvalue');
+            xmlWriter.writeAttributeString('event', setValue.event);
+            xmlWriter.writeAttributeString('ref', setValue.ref);
+            xmlWriter.writeAttributeString('value', setValue.value);
+            xmlWriter.writeEndElement();
+        }
+
+        _.each(form.getSetValues(), function (setValue) {
+            writeSetValue(setValue);
+        });
+
+        dataTree.walk(function (mug, nodeID, processChildren) {
+            if(mug && mug.options.getSetValues) {
+                _.each(mug.options.getSetValues(mug), function(setValue) {
+                    writeSetValue(setValue);
+                });
+            }
+            processChildren();
+        });
```

Comment:
Ah. From an earlier conversation I thought we didn't care about ordering at the moment.

Easy solution would be to use `vellum:ignore="retain"` if you really care.

We could add another input that had a priority associated with it.  The user could specify priority and we sort on that order. That sounds messy and hard to keep track of though, so I am against that. How common are problems relating  to the ordering of setvalues